README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
+
+```
+christie@win10 ~
+$ git clone git@github.com:YOURUSERNAME/zulip.git
+Cloning into 'zulip'...
+remote: Counting objects: 73571, done.
+remote: Compressing objects: 100% (2/2), done.
+remote: Total 73571 (delta 1), reused 0 (delta 0), pack-reused 73569
+Receiving objects: 100% (73571/73571), 105.30 MiB | 6.46 MiB/s, done.
+Resolving deltas: 100% (51448/51448), done.
+Checking connectivity... done.
+Checking out files: 100% (1912/1912), done.`
+```
+
+Now you are ready for [Step 3: Start the dev
+environment.](#step-3-start-the-dev-environment)
+
+### Step 3: Start the dev environment 
+
+Change into the zulip directory and tell vagrant to start the Zulip
+dev environment with `vagrant up`.
+
+```
+christie@win10 ~
+$ cd zulip
+
+christie@win10 ~/zulip
+$ vagrant up
+```
+
+The first time you run this command it will take some time because vagrant
+does the following:
+
+- downloads the base Ubuntu 14.04 virtual machine image (for OS X and Windows)
+  or container (for Ubuntu)
+- configures this virtual machine/container for use with Zulip,
+- creates a shared directory `/srv/zulip` within the virtual machine/container
+  that maps to where you have cloned the zulip code, and
+- runs the `provision.py` script, which downloads all required dependencies
+  and sets up the python environment for Zulip within the virtual
+  machine/container. 
+
+You will need an active internet connection during the entire processes.
+
+Once `vagrant up` has completed, connect to the dev environment with `vagrant
+ssh`:
+
+```
+christie@win10 ~/zulip
+$ vagrant ssh 
+```
+
+You should see something like this on Windows and OS X:
+
+```
+Welcome to Ubuntu 14.04.4 LTS (GNU/Linux 3.13.0-85-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com/
+
+  System information as of Wed May  4 21:45:43 UTC 2016
+
+  System load:  0.61              Processes:           88
+  Usage of /:   3.5% of 39.34GB   Users logged in:     0
+  Memory usage: 7%                IP address for eth0: 10.0.2.15
+  Swap usage:   0%
+
+  Graph this data and manage this system at:
+    https://landscape.canonical.com/
+
+  Get cloud support with Ubuntu Advantage Cloud Guest:
+    http://www.ubuntu.com/business/services/cloud
+
+0 packages can be updated.
+0 updates are security updates.
+```
+
+Or something as brief as this in the case of Ubuntu:
+
+```
+Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 4.4.0-21-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com/
+```
+
+Congrats, you're now inside the Zulip dev environment! 
+
+You can confirm this by looking at the command prompt, which starts with
+`(zulip-venv)`. 
+
+Next you'll navigate to the shared Zulip directory `/srv/zulip`, activate the
+python venv, and start the web server:
+
+``` 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:~
+$ cd /srv/zulip 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ source /srv/zulip-venv/bin/activate 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ ./tools/run-dev.py --interface='' 
```

Comment:
Note that the `source /srv/zulip-venv/bin/activate` line isn't necessary if the user is already in the venv.  This is because we've setup the `vagrant` user's `.bash_profile` to automatically enter the virtualenv, so that step is probably unnecessary.

I guess maybe this means we could simplify the whole thing to `/srv/zulip/tools/run-dev.py --interface=''`?